Task: Update Google Analytics ID

jonrandahl opened the pull request:
Alongside updating logging approach and migrating application to use Vite for Static Asset Management, this update also implements the Google Tag Manager ID to be loaded via a runtime variable as well as ensures there is consent before adding the GTM script to the application.

> [!NOTE]
> The `GOOGLE_ANALYTICS_ID` runtime variable will need to be added to the server configurations where applicable, absence of the variable will not break the application however reporting will not be triggered

This refactor also unifies the logging functionality across the application by creating a single logging module, replacing specific functions with general logging methods. 

It also updates dependencies in the Gemfile, adjusts paths within the application, and modifies the Makefile to streamline command execution.

## Changes

- Added configuration to accept the Google Tag Manager (GTM) ID as an runtime variable
- Ensured that the GTM scripts are not loaded without consent
- Introduced a unified logging module to standardise log requests.
- Refactored controller log calls to utilise the new logging module.
- Removed specific logging helper methods, consolidating into one module.
- Updated manifest icon paths for improved resource handling.
- Restructured application stylesheets for better maintainability.
- Adjusted the cookie handling logic to incorporate event listeners for interaction.
- Updated various gem dependencies, including a newer version of the logger.
- Simplified and streamlined the Makefile commands for consistency.
- Introduced new git hooks: post-commit, pre-commit, and pre-push.
- Pre-commit hook now lints Ruby files and resets if checks fail.
- Post-commit hook executes test suites and resets on failure.
- Pre-push hook validates Docker image builds and halts pushes if verification fails.
- Dockerfile simplified and modularized with multiple stages: alpine base, gems, npm, and build stages.
- Added new environment variables for Docker build customization.
- Updated logging behavior in controllers by simplifying log messages.
- Refactored stylesheet imports and structure, ensuring consistency and clarity.
- Replaced SCSS @import with @use, adhering to modern SCSS practices
- Removed legacy browser support, simplifying build configuration
- Added terser for JavaScript minification to improve performance
- Updated Sentry setup with systematic tag configuration
- Fixed issue with asynchronous (unnecessary) app version retrieval
- Removed deprecated govuk_template stylesheets

## Impact

- Accepting GTM ID as a variable allows future changes to be managed without complete update of codebase
- Ensuring acceptance of GTM adheres to current guidelines on cookie consent 
- Unified logging simplifies error monitoring and reduces code duplication across controllers.
- Dependency upgrades should improve application stability and performance.
- No breaking changes evident as most updates involve minor version changes—however, rigorous testing is recommended.
- Performance gains likely due to restructured styles and simplified logic in the logging and Makefile.
- Enhanced code quality and ensured strict adherence to testing standards as code must pass tests before being pushed.
- Avoided potential disruptions on key branches like 'hotfix', 'rebase', 'production' by skipping hooks.
- Removed reliance on obsolete webpack runners, reducing build complexity.
- Improved performance by optimizing Docker builds and allowing configuration flexibility with new ARGs and build stages.
- No breaking changes were introduced, maintaining backward compatibility.
- Behaviour now relies on modern browser features; some older browsers may no longer be supported
- SCSS maintains better namespace management with @use
- Simplified Sentry error tracking configuration

> [!IMPORTANT]
> No immediate breaking changes noticed, but older browser users may experience issues due to removing legacy support

jonrandahl committed:
chore: move font assets to JavaScript directory

- Moved font files from assets to JavaScript folder
- Organised fonts for better Vite integration

jonrandahl committed:
chore: move images to javascript directory

- Moved multiple image files from assets to javascript folder.
- Updated paths for favicons, logos, and markers.
- Maintained file integrity during the move.

jonrandahl committed:
chore: move stylesheets to JavaScript directory

All stylesheets have been relocated from the assets folder to the JavaScript directory. This includes various SCSS files, ensuring better organisation and accessibility within the project structure.

jonrandahl committed:
feat: add Vite setup and configuration

- Updated .gitignore to include Vite-related files
- Added a new bin script for running Vite
- Created Vite configuration file with development and test settings
- Introduced Procfile for managing development processes
- Set up Vite config with plugins for Rails, Vue, and Sentry

jonrandahl committed:
feat: update dependencies and add new gems

- Locked Rails to version 8.0 or higher
- Added `vite_rails` gem for Vite integration
- Removed `webpacker` dependency
- Included new gems: `csv`, `ostruct`
- Updated various other gem versions for compatibility and improvements
- Modified package.json to include new scripts and dependencies related to Vite

jonrandahl committed:
feat: replace HAML layout with ERB version

- Added a new ERB layout file.
- Removed old HAML layout files.
- Updated meta tags and scripts for better compatibility.
- Included conditional rendering for production environment features.

jonrandahl committed:
fix: Improve query result check

- Changed presence check to a more concise blank? method
- Simplified return logic for empty query results

jonrandahl committed:
feat(favicons): update favicon paths to use vite_asset_path

- Replaced image_path with vite_asset_path for favicons
- Removed several old apple-touch-icon links, keeping only the 180x180 version
- Updated manifest and mask-icon links to use vite_asset_path
- Added shortcut icon and msapplication-config meta tags

jonrandahl committed:
feat: update image tag to use vite_image_tag

Switched from `image_tag` to `vite_image_tag` for the logo. This improves asset handling and optimises loading.

jonrandahl committed:
feat: switch to vite_image_tag for logos

Updated image tags to use `vite_image_tag` instead of `image_tag`

- Changed logo rendering for multiple links in both Welsh and English versions
- Maintained alt text for accessibility

jonrandahl committed:
chore: remove unused asset and webpack config files

- Deleted asset initialiser file
- Removed development, production, and test webpack configs
- Cleared out loaders for ERB, Vue, and YAML

jonrandahl committed:
feat: remove lang attribute from template tags

Updated all components to remove the `lang='html'` attribute from `<template>` tags for consistency.

- Simplified template declarations across multiple files
- Ensured uniformity in component structure

jonrandahl committed:
chore: remove asset links from manifest

- Deleted image and CSS links
- Removed favicon configuration files

jonrandahl committed:
fix: update favicon paths to use vite_image_path

Updated the favicon manifest to use `vite_image_path` for image sources instead of `image_path`.

- Changed all icon source paths to use Vite helper
- Ensured consistency across different icon sizes

jonrandahl committed:
fix: update favicon paths to use vite_asset_path

- Changed manifest link to use vite_asset_path
- Ensured all favicon links are consistent with new asset path format

jonrandahl committed:
feat: update image tag for better performance

- Replaced `image_tag` with `vite_image_tag` for optimisation
- Maintained existing attributes like `srcset` and `alt`

jonrandahl committed:
chore: clean up asset config settings

Removed or commented out several asset-related configurations to finish removing webpacker

- Commented out sprockets railtie requirement
- Disabled asset debug mode
- Disabled quiet mode for assets
- Prevented fallback to assets pipeline in production

jonrandahl committed:
fix: update import path for VueI18n

- Changed the import path for VueI18n to a relative one
- Removed unused style block in data-view component
- Cleaned up imports in data-graph presenter

jonrandahl committed:
refactor: update import paths for consistency

- Changed relative imports to absolute imports for clarity
- Updated multiple components to use the new import style
- Maintained functionality while improving code readability

jonrandahl committed:
feat: update bundler script for better handling

Improved the bundler script to enhance functionality and error handling.

- Added methods for version management and environment variable checks.
- Implemented logic to determine the correct gemfile and lockfile paths.
- Enhanced error messages for failed bundler activation.
- Streamlined loading of the bundler based on command-line arguments.

jonrandahl committed:
feat: add Vue app entry point with Sentry integration

- Set up Vue with ElementUI and internationalisation
- Integrate Sentry for error tracking and session replay
- Configure locale settings for Numeral, moment.js, and D3.js
- Add debug logging for environment variables
- Implement IE Edge fix as per issue #169

jonrandahl committed:
build: add dotenv-rails and htmlbeautifier gems

- Added dotenv-rails for environment variable management in dev/test
- Replaced dotenvx with dotenv
- Included htmlbeautifier for HTML formatting tools

jonrandahl committed:
style: update ESLint config for dependencies

- Commented out the rule for extraneous dependencies
- Kept the debugger rule based on environment settings

jonrandahl committed:
feat: Add Procfile.dev for vite

- Commented out the web server command in the Procfile
- Kept vite command for development use

jonrandahl committed:
feat: update favicon manifest json

- Renamed the favicon manifest from ERB to JSON format
- Updated icon paths to relative URLs

jonrandahl committed:
style: reorganise SCSS imports for better structure

- Created a new SCSS file to manage imports.
- Moved third-party CSS imports to the new file.
- Sorted remaining stylesheets alphabetically in the new file.
- Deleted the old application.scss file.

jonrandahl committed:
feat: update vue config for asset URL handling

- Added transformAssetUrlsOptions to vue config
- Set includeAbsolute option to false

jonrandahl committed:
style: rename landing stylesheet and update image path

- Renamed landing.scss.erb to _landing.scss
- Changed background image path format

jonrandahl committed:
chore: update Gemfile for development group

- Moved 'haml-lint' and 'htmlbeautifier' to the development group
- Restored 'web-console' in the development group
- Cleaned up comments for clarity

jonrandahl committed:
chore: remove unused GOV.UK gems

- Deleted `govuk_elements_rails`, `govuk_frontend_toolkit`, and `govuk_template` from dependencies
- Cleaned up Gemfile.lock to reflect these removals

jonrandahl committed:
style: tidy up SCSS imports

- Added comments for clarity on helper functions and HTML elements
- Improved readability by formatting import statements

jonrandahl committed:
style: Update SCSS imports and clean up code

- Changed import paths to use aliases for better readability
- Added missing CSS files for element UI and Leaflet
- Removed unnecessary comments and unused imports
- Adjusted colour variables using new Sass functions
- Cleaned up the structure of various SCSS files for consistency

jonrandahl committed:
chore: remove dotenv-rails gem

Removed the dotenv-rails gem from the Gemfile and lock file.

- Cleaned up development and test group dependencies
- Updated related dependency lists in Gemfile.lock

jonrandahl committed:
chore: update dependencies and clean up package files

- Added new dependencies: `FontAwesome`, `govuk-elements-sass`, and `govuk_frontend_toolkit`.
- Updated Babel presets to the latest version.
- Removed unused dependencies from both regular and devDependencies.
- Cleaned up yarn.lock for better dependency management.

jonrandahl committed:
build: Update asset compilation process

- Changed bundle command to use variable for consistency
- Updated asset cleaning and compiling to use vite
- Added info command for vite output

jonrandahl committed:
build: Update Makefile commands for consistency

- Changed bundle and rails command invocations to use variables.
- Added a comment to clarify the removal of the GitHub token from bundle config.
- Ensured consistent usage of install with exec in lint and other tasks.

jonrandahl committed:
build: Update clean process and Gemfile handling

- Added a marker to the Gemfile for easier tracking
- Removed all assets using vite:clobber
- Cleared cache files from tmp/
- Cleaned up bundled files with force option
- Removed generated files including vendor, coverage, log, and node_modules
- Restored the original state of the Gemfile by removing the marker

jonrandahl committed:
fix: update layout for controllers to 'application'

- Changed layout from 'webpack_application' to 'application' in both browse and compare controllers.
- Ensured consistency across controller layouts.

jonrandahl committed:
style: Update GOV.UK stylesheets imports

Switched to using the GOV.UK elements from a new path.
- Replaced commented-out imports with active ones
- Maintained existing variable definitions for colours
- Organised imports for layout, typography, buttons, and forms

jonrandahl committed:
fix: update favicon paths to use images directory

- Changed favicon asset paths to point to the 'images' folder
- Updated all relevant link and meta tags for favicons

jonrandahl committed:
fix: update image paths for logos

- Changed logo image paths to include "images/" directory
- Updated alt text for accessibility in both Welsh and English versions

jonrandahl committed:
style: update stylesheet tag to SCSS format

- Changed stylesheet tag from 'application' to 'application.scss'
- Ensured media attribute remains set to 'all'

jonrandahl committed:
style: Update header styles for better clarity

- Adjusted spacing in the header selector
- Changed background colour to white for visibility

jonrandahl committed:
build: update `dotenv` import and config for dev

- Changed dotenv import to default import style
- Added loading of environment variables for development
- Kept production config intact

jonrandahl committed:
build: add dotenv and yaml plugin

- Added dotenv gem for environment variable management
- Included @modyfi/vite-plugin-yaml for YAML support
- Updated package.json to include dotenv and the new YAML plugin
- Enhanced Vite config with additional build options and plugins
- Updated Gemfile.lock and yarn.lock to reflect new dependencies

jonrandahl committed:
fix: update image path for header logo

- Changed image path to include 'images/' prefix
- Ensured consistency in header logo across views

jonrandahl committed:
style: reorganise stylesheet imports

Updated the order of stylesheet imports for better organisation and to avoid conflicts.

- Added utilities import at the top
- Maintained original order for specific styles
- Removed duplicates from previous list

jonrandahl committed:
feat: update locale imports for VueI18n

- Switched from require to import for locales
- Kept VueI18n usage intact
- Improved code consistency with ES6 syntax

jonrandahl committed:
refactor: - Removed client-side dotenv import as not running in node

- Updated comment to clarify purpose of the code block

jonrandahl committed:
refactor: update app version handling and Vue mount logic

- Added import for app version utility
- Updated current app release assignment to fallback on utility
- Changed debug logs to reflect new environment and version info
- Refactored Vue app mounting logic for better readability
- Removed unnecessary development environment check

jonrandahl committed:
feat: add version endpoint and app version retrieval

- Introduced a new endpoint to fetch the application version.
- Added a function to retrieve and store the current app version in session storage.
- Updated routing to include the new version route.

jonrandahl committed:
feat: add worker source to csp for Sentry integration

- Added `worker_src` directive for blob support to content security policy

jonrandahl committed:
chore: Update RuboCop gem requirements

- Changed 'rubocop' and 'rubocop-rails' to not require by default
- Removed duplicate 'rubocop' entry from development group

jonrandahl committed:
fix: update links to use relative URL root

Updated various links to use the application's relative URL root for better flexibility.

- Changed Welsh and English tool links in documentation
- Updated print layout navigation link

jonrandahl committed:
feat: add RAILS_ENV to development config

- Added RAILS_ENV variable set to 'development'

jonrandahl committed:
chore: tidy up component initialisation code

- Cleaned up unnecessary comments
- Improved readability of the component setup

jonrandahl committed:
feat: add GOV.UK template styles

Added styles for the GOV.UK templates and frontend toolkit.

- Imported GOV.UK template and accessibility styles manually
   - Created basic, header, footer, and styleguide files
- Updated Gemfile to include govuk_template gem
- Added necessary dependencies in package.json and yarn.lock

jonrandahl committed:
refactor: update application layout for better config

- Added Vite client tag for improved asset handling
- Set environment, version, locale, and root path in JS
- Included CSRF meta tags in production for security
- Cleaned up redundant code by removing duplicates

jonrandahl committed:
refactor: add conditional vue mount

- Updated Vue app mounting to only occur on specific routes
- Maintained existing i18n, store, and router setup
- Adjusted import order for Sentry and Vue

jonrandahl committed:
style: Suppress legacy colours warning temporarily

Added a variable to suppress the warning about using legacy organisation colours. This is just a temporary fix until those colours are phased out.

- Included comment explaining the purpose of the change
- Maintained existing imports without alterations

jonrandahl committed:
style: Remove GOV.UK frontend import

Removed the import of the GOV.UK Frontend Toolkit.
- Kept the suppressed warning variable for legacy colours.
- Cleaned up unnecessary code.

jonrandahl committed:
style: tidy up SCSS formatting

- Removed unnecessary comments for clarity
- Standardised text-size-adjust property across browsers
- Cleaned up whitespace in various sections

jonrandahl committed:
fix: improve query command handling

- Added logic to determine the appropriate query command based on user input.
- Enhanced ExplainQueryCommand with DataService inclusion and a new attribute for the query.
- Updated QueryCommand to indicate it is an explanation command.

jonrandahl committed:
fix: update qonsole path handling and refactor imports

- Added logic to use a local qonsole path in development
- Renamed import from VueI18n to i18n for consistency
- Adjusted visibility conditions for data view options and table

jonrandahl committed:
fix(router): update paths for relative URL handling

- Added workaround for Vue Router to manage relative URLs with Rails' asset pipeline.
- Updated route paths to use a variable for the relative URL root.
- Cleaned up code formatting by removing unnecessary line breaks.

jonrandahl committed:
fix: simplify data handling in server comms

- Destructured response to directly access sparql
- Added comment for clarity on fetching explanation condition

jonrandahl committed:
style: Update layout and styling for statistics view

- Changed display to block by default, flex on larger screens
- Adjusted checkbox container styles for better alignment
- Updated margin properties for checkboxes and images
- Improved responsiveness with media queries

jonrandahl committed:
build: Adds Yarn v4.8.1 release to the repository

- Includes core file system utilities.
- Introduces portable paths.

jonrandahl committed:
Build: Improves git ignore configuration

Enhances the .gitignore file to exclude more unnecessary files and directories.

This change updates the .gitignore file to:
- Ignore common dotfiles, bundler files, Ruby/Rails related files, RubyLSP config, and yarn files
- Improves the exclusion rules for log files, temporary files, node modules, and vite files.

jonrandahl committed:
Bump: Updates `data_services_api` gem

This resolves potential compatibility issues and incorporates new features from the updated dependency

jonrandahl committed:
Refactor: Moves marker assets to static imports

This change improves performance by removing the dependency on server-side rendering for asset paths and streamlines the asset loading process.

jonrandahl committed:
Chore: Removes default error pages

Removes the default error pages. This allows the application to handle error pages in a more flexible and customisable way.

jonrandahl committed:
Fix: Updates to Terser for ES6+ support

Updates Javascript compressor from Uglifier to Terser to better handle modern Javascript (ES6+)

- Bumps various gem dependencies.

jonrandahl committed:
Style: Enforces line length with rubocop

Enforces line length consistency in the content security policy configuration using rubocop.

jonrandahl committed:
Refactor: Updates cookie script to use Vite

Updates the cookie script to be handled by Vite instead of the asset pipeline.

*   Moves the cookie script to the javascript assets folder.
*   Updates the application layout to use `vite_javascript_tag` helper for the cookie script.

jonrandahl committed:
refactor: Fixes image path for data view statistics

- Resolves an issue where marker image paths were not correctly generated, leading to broken image links.
- Modifies the image source path function to correctly resolve image URLs using `import.meta.url`.
- Updates marker paths to reflect the location of the image assets.

jonrandahl committed:
Refactor: Use vite_asset_path for image asset

Updates the image path helper for the logo.

This ensures the correct asset path is generated when using Vite for asset management, improving the application's performance.

jonrandahl committed:
Merge branch 'task/release-candidate-v2.1.1' into spike/vite-upgrade-test

jonrandahl committed:
fix: Hides elements based on view state

Fixes a bug where certain elements were not being hidden correctly when the view state was present.

- Updates the visibility logic for non-JS and results table elements.
- Now hides these elements when the view state is available, ensuring consistent behaviour.

jonrandahl committed:
fix: Table cell alignment issue

Ensures that right-aligned table cells are not incorrectly centred.

jonrandahl committed:
fix: Resolves full image path retrieval

Updates the image path retrieval logic to return the pathname
instead of the complete URL.

This ensures correct image loading in different environments by
avoiding issues with URL construction.

jonrandahl committed:
feat: Adds legacy browser support

This improves compatibility for users on older systems by transpiling modern JavaScript features into code that can be executed in older environments.

- Updates dependencies to include the necessary polyfills and transforms.
- Configures the build process to generate the legacy-compatible code.

jonrandahl committed:
refactor: Use lr-common and element-rails

Updates stylesheets to use variables from `lr-common` and `element-rails`.

This change ensures consistent styling across the application by centralising common variables.

- Replaces hardcoded colour values with variables from `lr-common`
- Uses `element-rails` for colours related to the element UI library
- Makes table styling optional to prevent errors where it is not needed

jonrandahl committed:
refactor: Removes govuk_template stylesheets

These stylesheets are being removed to simplify the codebase.

jonrandahl committed:
fix: Configures Sentry tag for the application

The previous approach of setting the tag based on the environment
is replaced with a simpler solution. Now, the tag is set within
the initial scope of the Sentry initialisation.

jonrandahl committed:
fix: Resolves issue with app version retrieval

- Updates the app initialisation process to directly retrieve the version, streamlining the startup.

jonrandahl committed:
style: Adds terser to minify JavaScript

Adds terser as a dependency for vite to minify JavaScript files during the build process.

*   Improves page load times by reducing JavaScript bundle sizes.
*   Updates `yarn.lock` to reflect the new dependency and its sub-dependencies.

jonrandahl committed:
refactor: Updates SCSS imports to use `@use`

Updates SCSS imports to use the `@use` syntax instead of `@import`.

*   Improves modularity and avoids potential naming conflicts.
*   Reorders style imports for better consistency and to ensure dependencies are loaded correctly.

jonrandahl committed:
refactor: Removes legacy browser support

Removes the legacy browser support plugin.

This change streamlines the build process and reduces the size of the final bundle by removing code and dependencies related to supporting older browsers.

jonrandahl committed:
Merge remote-tracking branch 'origin/dev' into spike/vite-upgrade-test

jonrandahl committed:
refactor: update Google Analytics integration

- Remove direct script inclusion for analytics
- Add support for dynamic Google Analytics ID via env variables
- Enable analytics and debug logging in both production and debug modes
- Simplify HTML structure in views

jonrandahl committed:
build: update yarn to 4.9.2

jonrandahl committed:
chore: update json_rails_logger version

- Update dependency to latest patch version for stability

jonrandahl committed:
style: update rubocop configuration

- Set target Ruby version to 3.3 for improved compatibility
- Allow comment-style lines to exceed line length
- Exclude test files from some metrics checks
- Tune metrics to better guide code quality

jonrandahl committed:
refactor: improve logging and error handling

- Use more descriptive logging with structured data
- Simplify error logging and remove detailed log results
- Enhance exception handling with better debug information
- Adjust CSRF protection configuration for compatibility

jonrandahl committed:
refactor: unify logging functionality

- Replace existing logging helper with new module approach for consistency across apps
 - Refactor JSON formatting for all log messages
 - Use Rails logger for improved logging performance
 - Consolidate request logging behaviour to include more metadata
 - Optimise for reduced complexity and improved readability

jonrandahl committed:
refactor: standardise logging across controllers

- Replace existing logging helpers with consistent log method
- Enhance log messages with more descriptive content for clarity
- Improve internal monitoring and debugging process by unifying log format

jonrandahl committed:
refactor: streamline error logging

- Remove redundant error log type variable
- Consolidate error handling into unified methods
- Simplify log fields initialisation
- Ensure detailed error information is conditionally logged

jonrandahl committed:
refactor: streamline Makefile commands

- Remove unnecessary Gemfile modification logic in clean task
- Simplify docker run command with detached mode
- Consolidate start and server logic to remove redundancy
- Add version output command for better version tracking

jonrandahl committed:
chore: update gem dependencies

- Upgrade multiple gem dependencies to newer versions
- Improve compatibility and address potential security vulnerabilities
- Ensure continued support for existing features

jonrandahl committed:
refactor: simplify logging logic

- Remove redundant path logging
- Use dynamic service name in logs for consistency

jonrandahl committed:
refactor: adjust cookie handling logic

- Move cookie script to a different folder for better organisation
- Add event listeners for cookie banner buttons instead of inline JS
- Update analytics function to use const for variable declaration

jonrandahl committed:
refactor: restructure application styles

- Move application styles to a CSS entry point
- Rename and reorganise stylesheet imports

jonrandahl committed:
fix: update icon paths in manifest

- Remove relative paths from icon sources
- Ensure compatibility with path resolution

jonrandahl committed:
refactor: switch to haml for application layout

- Replace existing erb layout with haml format
- Retain existing meta and script tags with haml syntax
- Ensure compatibility with production environment scripts
- Simplify main content structure by removing redundant layers

jonrandahl committed:
chore: add local version for govuk_template

- Introduces a local version of gem files
- Sets initial version to 0.26.0

jonrandahl committed:
chore: add govuk template styles

- Integrate additional stylesheet for govuk template

jonrandahl committed:
refactor: restore layout structure

- Returns the appropriate selectors for layout
- Modify body class for consistency
- Add feedback request section under main content

jonrandahl committed:
chore: reconfigure GOV.UK stylesheets imports

- Add GOV.UK Frontend Toolkit and Elements stylesheets
- Suppress legacy organisation colour warnings temporarily
- Update configuration to optimize dependencies for GOV.UK toolkit

jonrandahl committed:
style: update layout for container and content as would not import

- Set max-width for container and content sections to improve responsiveness
- Adjust margins based on viewport width to enhance visual alignment
- Support better styling integration in the upcoming build improvements related to the project work in progress

jonrandahl committed:
chore: comment out manifest link in favicons

- Temporarily removes the manifest link for compatibility
- Simplifies the current setup without losing essential icons

jonrandahl committed:
style: restore button styling

- Add 'button' and 'button-start' classes to apply missing styles

jonrandahl committed:
style: update stylesheet imports for consistency

- Reorders CSS imports to avoid conflicts and improve clarity
- Restores additional stylesheets for UI components and mapping

jonrandahl committed:
style: update stylesheets for layout improvements

- Add flex layout for download links for better alignment
- Adjust header styles for consistent background color
- Add padding and media query adjustments for responsiveness
- Update button styles to enhance appearance and flexibility
- Migrate to using SCSS @import for stylesheet consistency

jonrandahl committed:
style: enhance download link styling

- fix consistent button styling to download links

jonrandahl committed:
chore: update dependency list

- Add new libraries to optimise dependency inclusion
- Include additional dependencies for current features

jonrandahl committed:
chore: simplify method by removing rubocop rule

- Removes the unnecessary size rule from headline summary

jonrandahl committed:
chore: remove obsolete webpack runners

- Remove old webpack and dev-server scripts
- Streamline build process by eliminating redundant scripts

jonrandahl committed:
build: update git hooks for automated checks

- Add post-commit hook to run tests and handle failures
- Modify pre-commit hook to skip specific branch checks
- Introduce pre-push hook to build and verify Docker images
- Allow skipping of hooks using the `--no-verify` flag
- Update README to provide setup instructions for hooks

jonrandahl committed:
chore: update Docker build steps in pre-push hook

- Refines Docker build process by utilizing AWS profiles
- Enhances profile handling with AWS_PROFILE environment variable
- Simplifies output messages during Docker image operations

jonrandahl committed:
refactor: adjust import path for marker routes

- Updates import declaration to use appropriate identifier
- Ensures URL generation for image paths uses correct source

jonrandahl committed:
refactor: simplify logging in controller

- Remove logging of request params and path to streamline logs
- Focus on essential logging for better performance

jonrandahl committed:
chore: update development dependencies

- Add new gems for language server and code intelligence
- Switch meta request to use a specific GitHub reference
- Update Gemfile.lock with newly added and updated gems
- Enhance code editors' functionality and code navigation

jonrandahl committed:
refactor: enhance Docker run configuration

- Add variable to manage Docker run flags for flexibility
- Incorporate Rails URL root configuration into Docker image build
- Ensure consistent use of Docker network creation and run options

jonrandahl committed:
chore: update favicon

- Removes and re-adds the favicon
- Ensures correct deployment of favicon

jonrandahl committed:
refactor: update Dockerfile to optimise build process

- Consolidate and streamline multiple build stages
- Use yarn for consistent dependency management
- Update to utilise vite for asset compilation
- Ensure minimal final image size and improve caching
- Enhance application security using defined user

jonrandahl committed:
build: disable asset inlining

- Prevents inlining of all assets for better performance
- Ensures sourcemaps are generated for use with Sentry

joescottdave commented on .eslintrc.js:
```diff
@@ -23,9 +23,9 @@ module.exports = {
-    }],
+    // 'import/no-extraneous-dependencies': ['error', {
+    //   'optionalDependencies': ['test/unit/index.js']
+    // }],
```
Can this come out?

joescottdave reviewed (approved):
Well done. Looks good.

One small question about a commented out rule

bogdanadrianmarc reviewed (approved):
A big one! LGTM, hopefully I didn't miss anything 👍

jonrandahl committed:
fix: correct bundle command usage

- Remove redundant bundle install command for config
- Ensure correct command format for bundle config setting

jonrandahl committed:
chore: clean up eslint config

- Remove commented-out rule regarding optional dependencies
